projects/element-ng/landing-page/login-basic/si-login-basic.component.ts

```diff
@@ -0,0 +1,203 @@
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SiLoginBasicComponent implements OnInit {
+  private static idCounter = 0;
+  /**
+   * Label for username input field.
+   *
+   * @defaultValue
+   * ```
+   * t(() => $localize`:@@SI_LOGIN_BASIC.USERNAME:Username`)
+   * ```
+   */
+  readonly usernameLabel = input(t(() => $localize`:@@SI_LOGIN_BASIC.USERNAME:Username`));
+  /**
+   * Label for password input field.
+   *
+   * @defaultValue
+   * ```
+   * t(() => $localize`:@@SI_LOGIN_BASIC.PASSWORD:Password`)
+   * ```
+   */
+  readonly passwordLabel = input(t(() => $localize`:@@SI_LOGIN_BASIC.PASSWORD:Password`));
+  /**
+   * Config for Forgot Password link.
+   */
+  readonly forgotPasswordLink = input<Link>();
+  /**
+   * Text for register now.
+   *
+   * @defaultValue
+   * ```
+   * t(() => $localize`:@@SI_LOGIN_BASIC.REGISTER_NOW_INTRO:Don't have an account?`)
+   * ```
+   */
+  readonly registerNowIntroText = input(
+    t(() => $localize`:@@SI_LOGIN_BASIC.REGISTER_NOW_INTRO:Don't have an account?`)
+  );
+  /**
+   * Config for Register Now link.
+   */
+  readonly registerNowLink = input<Link>();
+  /**
+   * Label for login button.
+   *
+   * @defaultValue
+   * ```
+   * t(() => $localize`:@@SI_LOGIN_BASIC.LOGIN:Login`)
+   * ```
+   */
+  readonly loginButtonLabel = input(t(() => $localize`:@@SI_LOGIN_BASIC.LOGIN:Login`));
+  /**
+   * Disables the login button.
+   *
+   * @defaultValue false
+   */
+  readonly disableLogin = input(false);
+  /**
+   * Enables the two-step login flow.
+   *
+   * @defaultValue false
+   */
+  readonly twoStep = input(false, { transform: booleanAttribute });
+
+  /**
+   * Indicates whether the login button should show a loading state.
+   *
+   * @defaultValue false
+   */
+  readonly loading = input(false);
+  /**
+   * Description of back button.
+   *
+   * @defaultValue
+   * ```
+   * t(() => $localize`:@@SI_LOGIN_BASIC.BACK:Back`)
+   * ```
+   */
+  readonly backButtonLabel = input(t(() => $localize`:@@SI_LOGIN_BASIC.BACK:Back`));
+  /**
+   * Description of next button.
+   *
+   * @defaultValue
+   * ```
+   * t(() => $localize`:@@SI_LOGIN_BASIC.NEXT:Next`)
+   * ```
+   */
+  readonly nextButtonLabel = input(t(() => $localize`:@@SI_LOGIN_BASIC.NEXT:Next`));
+  /**
+   * Emits username when user clicks on next button.
+   */
+  readonly usernameValidation = output<UsernameValidationPayload>();
+  /**
+   * Emits username and password on value change.
+   */
+  readonly valueChanged = output<UsernamePassword>();
+  /**
+   * Emits username and password on login event.
+   */
+  readonly login = output<UsernamePassword>();
+
+  private destroyRef = inject(DestroyRef);
+  protected readonly secondStepVisible = signal(false);
+  protected readonly loadingNext = signal(false);
+  protected usernameId = `__si-login-basic-username-${SiLoginBasicComponent.idCounter}`;
+  protected passwordId = `__si-login-basic-password-${SiLoginBasicComponent.idCounter++}`;
```

Comment:
```suggestion
  protected passwordId = `__si-login-basic-password-${SiLoginBasicComponent.idCounter++}`;
  protected usernameId = `__si-login-basic-username-${SiLoginBasicComponent.idCounter}`;
```

Otherwise we have different ID counter

Reply:
I was thinking we could also do something like this to keep both IDs on the same index, which looks a bit cleaner to me:
```ts
private readonly index = SiLoginBasicComponent.idCounter++;
protected usernameId = `__si-login-basic-username-${this.index}`;
protected passwordId = `__si-login-basic-password-${this.index}`;
```
What do you think?

Reply:
yes, even better